feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -285,7 +272,49 @@ class BillingWrapper(
         )
     }
 
-    fun consumePurchase(
+    override fun consumeAndSave(
+        shouldTryToConsume: Boolean,
+        purchase: PurchaseWrapper
+    ) {
+        if (purchase.type == ProductType.UNKNOWN) {
+            // Would only get here if the purchase was triggered from outside of the app and there was
+            // an issue getting the purchase type
+            return
+        }
+        if (purchase.purchaseState == RevenueCatPurchaseState.PENDING) {
+            // PENDING purchases should not be fulfilled
```

Comment:
The thing here is that when `consumeAndSave` is called with a PurchaseWrapper that comes from a restore, we don't know the state of the purchase. There is no purchase state in the `PurchaseHistoryRecord` object from Google. 

I would have to test if `queryPurchaseHistoryAsync` can return `PENDING` purchases (this open issue says it does https://github.com/android/play-billing-samples/issues/367).

Anyhow, in the case that the purchase returned by `queryPurchaseHistoryAsync` was in a Pending state, and we try to consuming/acknowledge it in this function, consuming/acknowledging would fail and we would just log it.

I think it's safer to initialize PurchaseWrappers that come from a restore with a `RevenueCatPurchaseState.UNSPECIFIED_STATE` and make sure we exit from this function if the state is `RevenueCatPurchaseState.PENDING` and make this comment say `PENDING purchases should not be fulfilled`. I am worried of new states added to the enum though 🤔

Reply:
Maybe we should just assume all purchases returned from `queryPurchaseHistoryAsync` are `PURCHASED`